biz.aQute.bndlib/src/aQute/bnd/build/model/BndEditModel.java

```diff
@@ -677,12 +685,90 @@ public void setIncludeSources(boolean includeSources) {
 	}
 
 	public List<String> getPrivatePackages() {
-		return doGetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, listConverter);
+		List<String> privatePackagesEntries1 = getEntries(Constants.PRIVATEPACKAGE, listConverter);
+		List<String> privatePackagesEntries2 = getEntries(Constants.PRIVATE_PACKAGE, listConverter);
+
+		Set<String> privatePackages = Stream.concat(privatePackagesEntries1.stream(), privatePackagesEntries2.stream())
+			.collect(toSet());
+		return new ArrayList<>(privatePackages);
+	}
+
+	public void setPrivatePackages(List<String> newPackages) {
+		List<String> privatePackagesEntries1 = getEntries(Constants.PRIVATEPACKAGE, listConverter);
+		List<String> privatePackagesEntries2 = getEntries(Constants.PRIVATE_PACKAGE, listConverter);
+		
+		Set<String> privatePackages = Stream.concat(privatePackagesEntries1.stream(), privatePackagesEntries2.stream())
+			.collect(toSet());
+		
+		List<String> addedEntries = disjunction(newPackages, privatePackages);
+		List<String> removedEntries = disjunction(privatePackages, newPackages);
+
+		privatePackagesEntries1.removeAll(removedEntries);
+		if (privatePackagesEntries1.isEmpty()) {
+			removeEntries(Constants.PRIVATEPACKAGE);
+		} else {
+			setEntries(privatePackagesEntries1, Constants.PRIVATEPACKAGE);
+		}
+
+		privatePackagesEntries2.removeAll(removedEntries);
+		if (privatePackagesEntries2.isEmpty()) {
+			removeEntries(Constants.PRIVATE_PACKAGE);
+		} else {
+			setEntries(privatePackagesEntries2, Constants.PRIVATE_PACKAGE);
+		}
+		
+		if (hasPrivatePackageInstruction()) {
+			privatePackagesEntries1.addAll(addedEntries);
+			setEntries(privatePackagesEntries1, Constants.PRIVATEPACKAGE);
+		} else {
+			privatePackagesEntries2.addAll(addedEntries);
+			setEntries(privatePackagesEntries2, Constants.PRIVATE_PACKAGE);
+		}
+	}
+
+	public void setEntries(List<? extends String> packages, String key) {
+		List<String> oldPackages = getEntries(key, listConverter);
+		doSetObject(key, oldPackages, packages, stringListFormatter);
+	}
+
+	public void removeEntries(String key) {
+		List<String> oldPackages = getEntries(key, listConverter);
+		doRemoveObject(key, oldPackages, null, stringListFormatter);
 	}
 
-	public void setPrivatePackages(List<? extends String> packages) {
-		List<String> oldPackages = getPrivatePackages();
-		doSetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, oldPackages, packages, stringListFormatter);
+	public void addPrivatePackage(String packageName) {
+		String key = hasPrivatePackageHeader() ? Constants.PRIVATE_PACKAGE : Constants.PRIVATEPACKAGE;
```

Comment:
In setPrivatePackages, you prefer the instruction. Here you prefer the header. I think we should prefer the newer instruction. If you use the instruction, we add new things to it.

Reply:
Agreed and done